### PR TITLE
Jest util tests

### PR DESCRIPTION
**Pull request by m-allanson**
> Which issue, if any, is this issue related to?

Refs #2083
Replaces #2085

> Is there anything in the PR that needs further explanation?

This should fix the recurring issues noted in #2085. If we're sticking with tape then this PR can just be closed.

### CODE REVIEW

**Comment by davidtheclark** [comment]
Awesome work. Too bad it seems to be hitting the same problems noted in #2131.

File: src/utils/__tests__/findAtRuleContext.test.js [hunk 1]
```diff
@@ -14,21 +13,20 @@ test("findAtRuleContext", t => {
     d {}
   `
 
-  t.plan(4)
   postcss().process(css).then(result => {
```
Comment (davidtheclark) [inline src/utils/__tests__/findAtRuleContext.test.js]: Should return this Promise, I think.

**Comment by davidtheclark** [comment]
> If we're sticking with tape then this PR can just be closed.

@m-allanson Yeah, I'm afraid we have to see if the Jest issues can be resolved first ... let's circle back to this within a week or two to see if we think it's a good idea or not.

**Comment by jeddy3** [comment]
The current CI performance for using jest for `lib`, `utils` and `system-tests` looks good in this branch (using the `-i` flag), and that's still with babel in the mix. 

I think it could be worth, irrespective of how jest pans out for rule tests, continuing with this initiative and moving the `utils` tests to jest once the `drop-babel` branch is merged. This feels like it might give the best balance between DX and CI performance.

**Comment by davidtheclark** [comment]
I will try to hash this out tonight.

**Comment by davidtheclark** [comment]
I agree with @jeddy3 that we can merge this as long as we can get it consistently passing on CI. Probably the only significant work that will need to be done after the Great Babel Removal is to change `import` and `export` to CommonJS.

**Comment by m-allanson** [comment]
I can take a look at rebasing this tonight (GMT).

**Pull request closed without merging**